test/unit/math/rev/arr/functor/integrate_ode_rk45_test.cpp

```diff
@@ -116,3 +120,153 @@ TEST(StanAgradRevOde_integrate_ode_rk45, lorenz_finite_diff) {
 
   test_ode(lorenz, t0, ts, y0, theta, x, x_int, 1e-8, 1e-1);
 }
+
+TEST(StanAgradRevOde_integrate_ode_rk45, time_steps_as_param) {
+  using stan::math::integrate_ode_rk45;
+  using stan::math::to_var;
+  using stan::math::value_of;
+
+  const double t0 = 0.0;
+  forced_harm_osc_ode_fun ode;
+  std::vector<double> theta{0.15, 0.25};
+  std::vector<double> y0{1.0, 0.0};
+  std::vector<stan::math::var> ts;
+  for (int i = 0; i < 100; i++)
+    ts.push_back(t0 + 0.1 * (i + 1));
+  std::vector<double> x;
+  std::vector<int> x_int;
+  std::vector<stan::math::var> y0v = to_var(y0);
+  std::vector<stan::math::var> thetav = to_var(theta);
+  stan::math::var t0v = 0.0;
+
+  std::vector<std::vector<stan::math::var>> res;
+
+  std::vector<std::vector<double>> res_d
+      = integrate_ode_rk45(ode, y0, t0, value_of(ts), theta, x, x_int);
+
+  // here we only test first & last steps, and rely on the
+  // fact that results in-between affect the initial
+  // condition of the last step to check their validity.
```

Comment:
I think that's fine, though it shouldn't be too difficult to write the test and loop through all the states to make sure the correct answer is always attained.

Reply:
Our grad tests for the ode solvers essentially work the same way as this one.

Reply:
Ok, just a suggestion. Though I wonder if there was an initial motivation for doing this. Maybe cutting down the time of the unit test.